feat(contrib/linode): add Linode provision scripts and docs

croemmich opened the pull request:
Linode's recent switch to KVM gave me an added push to rewrite our team's provisioning tools to work with the new hypervisor as it makes running CoreOS much simpler. This time, I did so with the intention of fully documenting the process for the community and our team at Myriad Mobile as well as make our tools publicly available. So here's a stab at that!

carmstrong commented:
This is awesome, @croemmich!! We'll give it a review!

carmstrong commented on contrib/linode/linode-user-data-template.yaml:
```diff
@@ -0,0 +1,46 @@
+      [Service]
+      Type=oneshot
+      ExecStart=/bin/sh -c "sysctl -w net.ipv6.conf.all.disable_ipv6=1"
+    # use ntpd rather than timesyncd. timesyncd was not keeping the nodes close enough to keep ceph happy
```
Perhaps we should file this issue upstream. I'm sure the CoreOS team would want to know that this change had this effect... the time sync daemon was changed in CoreOS 681: https://coreos.com/releases/#681.0.0

croemmich replied:
Agreed, I've submitted https://github.com/coreos/bugs/issues/391. Unfortunately systemd-timesyncd doesn't seem heavily documented or mature :disappointed:

mboersma commented on docs/installing_deis/linode.rst:
```diff
@@ -0,0 +1,218 @@
+-------------------------
+
+Please refer to :ref:`system-requirements` for resource considerations when choosing a Linode
+plan to run Deis. The smallest recommended cluster consists of 3 Linode 4096s.
```
Could you replace with or add this wording? We just added this to the other "Installing" docs for consistency:
```
A Deis cluster must have 3 or more nodes. See :ref:`cluster-size` for more details.
```

mboersma commented:
This is fantastic. The python could use some comments but is well-crafted and easy to follow. Code LGTM.

bacongobbler commented:
I used to have a Ramnode account, but no Linode account. Sorry!

croemmich commented:
@mboersma I've updated the docs to include your wording. Additionally I have added validation to the provision script to ensure the specified Linode plan and node number is consistent with the system requirements.

carmstrong commented:
I think this is fine to :ship: since @croemmich mentioned in the docs that it's supported by our community. Great work! Code LGTM.

mboersma commented:
This passed the only relevant tests: the docs. Nothing under `/contrib` is tested by CI. Merging...
